.github/workflows/build.yml

```diff
@@ -411,18 +411,21 @@ jobs:
 
   build_asan:
     name: 'Address sanitizer'
-    runs-on: ubuntu-22.04
+    runs-on: ${{ matrix.os }}
     timeout-minutes: 60
     needs: check_source
     if: needs.check_source.outputs.run_tests == 'true'
+    strategy:
+      matrix:
+        os: [ubuntu-22.04]
```

Comment:
The change itself looks ok, but I was wondering if it was just for consistency or if there is another reason.

Reply:
I think it's because otherwise the `matrix.os` variable here is undefined: https://github.com/python/cpython/blob/c5b99f5c2c5347d66b9da362773969c531fb6c85/.github/workflows/build.yml#L449

Reply:
That's right.

This was added in https://github.com/python/cpython/pull/124403, where we replaced `${{ runner.os }}` (which evaluates to `Linux`) with `${{ matrix.os }}` (which evaluates to `ubuntu-22.04`), so the cache would be cleared when we update to 24.04.

In that PR, you can see we added `os:` to the top one, but forgot for the bottom one (and we didn't have a linter).

See also https://github.com/python/cpython/pull/123699#issuecomment-2351821568 for more background, cc @zware.